flink-python/pyflink/ml/api/param/base.py

```diff
@@ -0,0 +1,357 @@
+        :return: the type of the specific parameter.
+        """
+        self._param_map.pop(info)
+
+    def contains(self, info: 'ParamInfo') -> bool:
+        """
+        Check whether this params has a value set for the given `info`.
+
+        :param info: the info of the specific parameter to check.
+        :return: `True` if this params has a value set for the specified `info`, false otherwise.
+        """
+        return info in self._param_map
+
+    def size(self) -> int:
+        """
+        Return the number of params.
+
+        :return: Return the number of params.
+        """
+        return len(self._param_map)
+
+    def clear(self) -> None:
+        """
+        Removes all of the params. The params will be empty after this call returns.
+
+        :return: None.
+        """
+        self._param_map.clear()
+
+    def is_empty(self) -> bool:
+        """
+        Returns `true` if this params contains no mappings.
+
+        :return: `true` if this params contains no mappings.
+        """
+        return len(self._param_map) == 0
+
+    def to_json(self) -> str:
+        """
+        Returns a json containing all parameters in this Params. The json should be
+        human-readable if possible.
+
+        :return: a json containing all parameters in this Params.
+        """
+        import jsonpickle
+        return str(jsonpickle.encode(self._param_map, keys=True))
+
+    def load_json(self, json: str) -> 'Params':
+        """
+        Restores the parameters from the given json. The parameters should be exactly
+        the same with the one who was serialized to the input json after the restoration.
+
+        :param json: the json String to restore from.
+        :return: the Params.
+        """
+        import jsonpickle
+        self._param_map.update(jsonpickle.decode(json, keys=True))
+        return self
+
+    @staticmethod
+    def from_json(json) -> 'Params':
+        """
+        Factory method for constructing params.
+
+        :param json: the json string to load.
+        :return: the `Params` loaded from the json string.
+        """
+        return Params().load_json(json)
+
+    def merge(self, other_params: 'Params') -> 'Params':
+        """
+        Merge other params into this.
+
+        :param other_params: other params.
+        :return: return this Params.
+        """
+        if other_params is not None:
+            self._param_map.update(other_params._param_map)
+        return self
+
+    def clone(self) -> 'Params':
+        """
+        Creates and returns a deep clone of this Params.
+
+        :return: a clone of this Params.
+        """
+        new_params = Params()
+        new_params._param_map.update(self._param_map)
+        return new_params
+
+
+class ParamInfo(object):
+    """
+    Definition of a parameter, including name, description, type_converter and so on.
+    """
+
+    def __init__(self, name, description, is_optional=True,
```

Comment:
one thing I notice missing comparing to java is `alias`. do we plan to support that?

Reply:
Good catch. I have not added this due to something that confused me. In Java, the `ParamInfo` provides an alias for each paraminfo, but whenever set a paraminfo, only the name is used, see `Params.set()`. It seems we don't need this member. What do you think?

Reply:
I share the same confusion. my intuition is: if it is not used, let's do not just add a placeholder for the sake of interface parity. +1 to keep the current `__init__` interface.